sbi/inference/posteriors/sbi_posterior.py

```diff
@@ -43,7 +49,8 @@ def __init__(
             neural_net: A classifier for sre/aalr, a density estimator for snpe/snl.
             prior: Prior distribution with methods `log_prob` and `sample`.
             x_o: Observation acting as conditioning context. Absent if None.
-            sample_with_mcmc: Whether to sample with MCMC to correct leakage.
+            sample_with_mcmc: Whether to sample with MCMC. Will always be `True` for SRE
+                and SNL, but can also be set to `True` for SNPE to deal with leakage.
```

Comment:
"but can also be set to `True` for SNPE to use MCMC for dealing with leakage."

Reply:
+1 to 'to use MCMC', but please don't change to gerund ('for dealing') ;).

Also, I think what you want to say with 'will always be true' is 'your setting will be ignored if using SRE/SNL, it needs to be True.'

Reply:
In which case, shouldn't it be just a flag about correcting for leakage _in snpe_ _with mcmc_?